File: solutions/zig/sublist/1/sublist.zig
const std = @import("std");

pub const Relation = enum {
    equal,
    sublist,
    superlist,
    unequal,
};

fn isSublist(list_one: []const i32, list_two: []const i32) bool {
    if (list_one.len == 0) {
        return true;
    }
    if (list_one.len > list_two.len) {
        return false;
    }
    for (0..(list_two.len - list_one.len + 1)) |i| {
        if (std.mem.eql(i32, list_one, list_two[i..i+list_one.len])) {
            return true;
        }
    }
    return false;
}

pub fn compare(list_one: []const i32, list_two: []const i32) Relation {
    if (std.mem.eql(i32, list_one, list_two)) {
        return .equal;
    }
    if (isSublist(list_one, list_two)) {
        return .sublist;
    }
    if (isSublist(list_two, list_one)) {
        return .superlist;
    }
    return .unequal;
}

File: solutions/php/sublist/1/Sublist.php
<?php
declare(strict_types=1);

class Sublist
{
    function compare(array $listOne, array $listTwo): string
    {
        if ($this->isEqual($listOne, $listTwo)) {
            return 'EQUAL';
        }
        if ($this->isSublist($listOne, $listTwo)) {
            return 'SUBLIST';
        }
        if ($this->isSuperlist($listOne, $listTwo)) {
            return 'SUPERLIST';
        }
        return 'UNEQUAL';
    }

    private function isEqual(array $listOne, array $listTwo): bool
    {
        if (count($listOne) !== count($listTwo)) {
            return false;
        }
        for ($i = 0; $i < count($listOne); $i++) {
            if ($listOne[$i] !== $listTwo[$i]) {
                return false;
            }
        }
        return true;
    }

    private function isSublist(array $listOne, array $listTwo): bool
    {
        if (count($listOne) >= count($listTwo)) {
            return false;
        }
        for ($i = 0; $i <= count($listTwo) - count($listOne); $i++) {
            $isSub = true;
            for ($j = 0; $j < count($listOne); $j++) {
                if ($listTwo[$i + $j] !== $listOne[$j]) {
                    $isSub = false;
                    break;
                }
            }
            if ($isSub) {
                return true;
            }
        }
        return false;
    }

    private function isSuperlist(array $listOne, array $listTwo): bool
    {
        return $this->isSublist($listTwo, $listOne);
    }
}
